### DISCUSSION TITLE
Kernel dies when plotting Mesh extracted from GemPy with GemGIS

### DISCUSSION THREAD

**Discussion by AlexanderJuestel** [pyvista/pyvista#4596] [Q&A] [answered]
Hello, 

I recently noticed that my GemGIS function `create_depth_maps_from_gempy` (https://github.com/cgre-aachen/gemgis/blob/f9a8afebe9bef388bb78f34c80af495bed7a6e97/gemgis/visualization.py#L1521-L1617) that creates a PyVista PolyData set from a GemPy surface fails with the following error in line `surf = pv.PolyData(val['vertices'], np.insert(val['edges'], 0, 3, axis=1).ravel())` because of the dimensions of `val['vertices']`. This error was not encountered before:

`ValueError: Dimension of ``points`` should be 1 or 2, not 3`

`val['vertices']` is a list containing one element, an array. 

The error can therefore be solved using `val['vertices'][0]` instead, which has then the correct shape with N,3 as in the docs and a mesh will be returned. However, when trying to plot this mesh, the kernel dies every single time. I have attached the mesh that fails. We could reproduce with different PyVista versions and on another Windows machine. So the question is if something goes wrong during the creation of the meshes....The version with `val['vertices']` is still implemented in GemPy and still seems to work so I really wonder what I messed up here....

Thanks for the support. 
Cheers
Alex

<html>
<body>

Wed Jun 28 11:40:10 2023 Mitteleuropäische Sommerzeit
--
OS | Windows | CPU(s) | 16 | Machine | AMD64
Architecture | 64bit | RAM | 15.2 GiB | Environment | Jupyter
GPU Vendor | ATI Technologies Inc. | GPU Renderer | AMD Radeon(TM) Graphics | GPU Version | 4.5.14761 Core Profile Context 21.30.40.09 30.0.13040.9001
MathText Support | True
Python 3.10.11 \| packaged by Anaconda, Inc. \| (main, May 16 2023, 00:55:32) [MSC v.1916 64 bit (AMD64)]
pyvista | 0.39.1 | vtk | 9.2.6 | numpy | 1.25.0
matplotlib | 3.7.1 | scooby | 0.7.2 | pooch | v1.7.0
imageio | 2.31.1 | pyvistaqt | 0.10.0 | PyQt5 | 5.15.9
IPython | 8.14.0 | ipywidgets | 8.0.6 | scipy | 1.11.0
tqdm | 4.65.0 | meshio | 4.4.6 | nest_asyncio | 1.5.6

</body>
</html>

Full Error Message:

```---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[27], line 1
----> 1 mesh = gg.visualization.create_depth_maps_from_gempy(geo_model, surfaces=['Fault'])
      2 mesh

File ~\Anaconda3\envs\gempy2.3\lib\site-packages\gemgis\visualization.py:1606, in create_depth_maps_from_gempy(geo_model, surfaces)
   1602 surfaces_poly = {}
   1604 for idx, val in geometric_data[['vertices', 'edges', 'color', 'surface', 'id']].dropna().iterrows():
   1605     # Creating PolyData from each surface
-> 1606     surf = pv.PolyData(val['vertices'], np.insert(val['edges'], 0, 3, axis=1).ravel())
   1608     # Extract depth data for each surface
   1609     depth = geometric_data['vertices'][geometric_data[geometric_data['surface'] == val['surface']].index[0]][:, 2]

File ~\Anaconda3\envs\gempy2.3\lib\site-packages\pyvista\core\pointset.py:624, in PolyData.__init__(self, var_inp, faces, n_faces, lines, n_lines, strips, n_strips, deep, force_ext, force_float)
    622 # First parameter is points
    623 if isinstance(var_inp, (np.ndarray, list, _vtk.vtkDataArray)):
--> 624     self.SetPoints(pyvista.vtk_points(var_inp, deep=deep, force_float=force_float))
    626 else:
    627     msg = f"""
    628         Invalid Input type:
    629 
   (...)
    636 
    637         Instead got: {type(var_inp)}"""

File ~\Anaconda3\envs\gempy2.3\lib\site-packages\pyvista\utilities\helpers.py:519, in vtk_points(points, deep, force_float)
    517     points = points.reshape(-1, 3)
    518 elif points.ndim > 2:
--> 519     raise ValueError(f'Dimension of ``points`` should be 1 or 2, not {points.ndim}')
    521 # verify shape
    522 if points.shape[1] != 3:

ValueError: Dimension of ``points`` should be 1 or 2, not 3
```

[Test.zip](https://github.com/pyvista/pyvista/files/11892631/Test.zip)

**Comment by banesullivan** [comment, marked as answer]
It looks like the cell connectivity is malformed. Something is probably awry with `np.insert(val['edges'], 0, 3, axis=1).ravel()`.

One key to this is the stark difference in `n_points` (315) and `n_cells` (26). If they are all triangle cells, this does not align. Similar for line cells or quad cells.

**Reply by AlexanderJuestel** [reply]
You were right @banesullivan! It was as easy as adding a `[0]` to after `val['edges']` and everything works perfectly again....

![image](https://github.com/pyvista/pyvista/assets/45469915/1fe019ed-9441-4cbb-b8f7-8d886b5559d0)
